operator/v1/types_ingress.go

```diff
@@ -229,6 +236,138 @@ var (
 	DNSReadyIngressConditionType = "DNSReady"
 )
 
+// SecuritySpec defines the settings for securing IngressController connections.
+type SecuritySpec struct {
```

Comment:
I still question whether this conforms to union type conventions and wish there was some response re: the proposed organization and naming of fields (https://github.com/openshift/api/pull/322#issuecomment-500495748)